src/components/Gradebook/index.jsx

```diff
@@ -48,8 +52,16 @@ export default class Gradebook extends React.Component {
     this.props.initializeFilters(urlQuery);
     this.props.getRoles(this.props.courseId);
     this.overrideReasonInput.focus();
-    const { assignmentGradeMin, assignmentGradeMax } = urlQuery;
-    this.setState({ assignmentGradeMin, assignmentGradeMax });
+
+    // do not override the default values if the query parameters are undefined
+    const courseGradeMin = urlQuery.courseGradeMin || this.state.courseGradeMin;
```

Comment:
I realize that this has the potential to set the state to the exact same state as before. I know React wold be smart enough to not actually re-render (not sure about the virtual DOM), so I'm not sure if it's an issue or not.

Reply:
I would recommend building the object you're going to pass to setState imperatively (I know, ew) a la
```
const newStateFields = {};
['assignmentGradeMin', ... , 'courseGradeMax'].forEach(attr => {
  if (urlQuery[attr]) {
    newStateFields[attr] = urlQuery[attr];
  }
});
this.setState(newStateFields);
```

If you wanted to write this fully functionally, you'd need to use [the functional form of setState](https://stackoverflow.com/a/48209870) since the new version depends on the previous values of state.